warp10/src/main/java/io/warp10/script/functions/GOLDWRAP.java

```diff
@@ -0,0 +1,171 @@
+              bytes = OrderPreservingBase64.decode(element.toString().getBytes(Charsets.US_ASCII));
+            } else {
+              bytes = (byte[]) element;
+            }
+            GTSWrapper wrapper = new GTSWrapper();
+            deser.deserialize(wrapper, bytes);
+            encoder = GTSWrapperHelper.fromGTSWrapperToGTSEncoder(wrapper);      
+          } else {
+            throw new WarpScriptException(getName() + " can only be applied to Geo Time Series™, GTS Encoders or wrapped instances of those types.");
+          } 
+          
+          //
+          // Split the encoder in 5 GTS, one per type, in this order:
+          //
+          // LONG, DOUBLE, BOOLEAN, STRING, BINARY
+          //
+          
+          GeoTimeSerie[] gts = new GeoTimeSerie[5];
+
+          for (int i = 0; i < gts.length; i++) {
+            gts[i] = new GeoTimeSerie();
+          }
+          
+          GTSDecoder decoder = encoder.getDecoder();
+          
+          // Populate the 5 GTS
+          while(decoder.next()) {
+            long ts = decoder.getTimestamp();
+            long location = decoder.getLocation();
+            long elevation = decoder.getElevation();
+            Object value = decoder.getBinaryValue();
+            
+            if (value instanceof Long) {
+              GTSHelper.setValue(gts[0], ts, location, elevation, value, false);
+            } else if (value instanceof Double || value instanceof BigDecimal) {
+              GTSHelper.setValue(gts[1], ts, location, elevation, value, false);          
+            } else if (value instanceof Boolean) {
+              GTSHelper.setValue(gts[2], ts, location, elevation, value, false);
+            } else if (value instanceof String) {
+              GTSHelper.setValue(gts[3], ts, location, elevation, value, false);
+            } else if (value instanceof byte[]) {
+              GTSHelper.setValue(gts[4], ts, location, elevation, value, false);
+            }
+          }
+          
+          // Sort the 5 GTS using fullsort so we get a deterministic order
+          // in the presence of duplicate ticks
+          
+          for (int i = 0; i < gts.length; i++) {
+            GTSHelper.fullsort(gts[i], false);
+          }
+          
+          // Now merge the GTS in time order with the type precedence of the 'gts' array
+          
+          GTSEncoder enc = new GTSEncoder(0L);
+          enc.setMetadata(encoder.getMetadata());
+          
+          int[] idx = new int[gts.length];
+          
+          while (true) {
+            // Determine the next GTS to add from its timestamp, lowest first
+            int gtsidx = -1;
+            
+            long ts = Long.MAX_VALUE;
+            for (int i = 0; i < gts.length; i++) {
+              if (idx[i] >= GTSHelper.nvalues(gts[i])) {
+                continue;
+              }
+              long tick = GTSHelper.tickAtIndex(gts[i], idx[i]);
+              if (tick < ts) {
```

Comment:
Better use `-1 == gtsidx || tick < ts` to avoid missing a tick at `Long.MAX_VALUE`